internal/extsvc/bitbucketserver/client.go

```diff
@@ -779,6 +812,98 @@ type PullRequest struct {
 			Href string `json:"href"`
 		} `json:"self"`
 	} `json:"links"`
+
+	Activities []Activity `json:"activities,omitempty"`
+}
+
+// Activity is a union type of all supported pull request activity items.
+type Activity struct {
+	ID          int            `json:"id"`
+	CreatedDate int            `json:"createdDate"`
+	User        User           `json:"user"`
+	Action      ActivityAction `json:"action"`
+
+	// Comment activity fields.
+	CommentAction string         `json:"commentAction,omitempty"`
+	Comment       *Comment       `json:"comment,omitempty"`
+	CommentAnchor *CommentAnchor `json:"commentAnchor,omitempty"`
+
+	// Reviewers change fields.
+	AddedReviewers   []User `json:"addedReviewers,omitempty"`
+	RemovedReviewers []User `json:"removedReviewers,omitempty"`
+
+	// Merged event fields.
+	Commit *Commit `json:"commit,omitempty"`
+}
+
+// ActivityAction defines the action taken in an Activity.
+type ActivityAction string
+
+// Known ActivityActions
+const (
+	ApprovedActivitiyAction   ActivityAction = "APPROVED"
```

Comment:
There's a typo in all of these `*Action` `const`s:

```suggestion
	ApprovedActivityAction   ActivityAction = "APPROVED"
```